File: src/python/factorial.py
""" Iterative factorial algorithm """


def factorial(num):
    """
    Implementation of the factorial algorithm iteratively.

    Arguments:
        num: int. the number of which you want to obtain the factorial.

    Returns the result of the factorial operation
    """
    aux = 1
    for i in range(2, num + 1):
        aux = aux * i
    return aux


if __name__ == "__main__":
    print(factorial(5))

File: src/c/Factorial.c
#include <stdio.h>
#include <stdlib.h>

/*
 *	Examples of functions to return the factorial of a number n
 *	função recursiva
 */

int main() {
  int num;
  printf("Digite um número: ");
  scanf("%d", &num);
  int result = fatorial(num);
  printf("1 => ");
  printf("%d! é : %d\n", num, result);
  return (0);
}

int fatorial(int num) {
  if (num <= 1) {
    return 1;
  }
  printf("%d * ", num);
  return num * fatorial(num - 1);
}
